File: app-back/src/models/cityModel.ts
import { ProvinceModel } from "./provinceModel";

export class CityModel {
  id: number;
  name: string;
  provinceId: number;
  province?: ProvinceModel;

  constructor(id: number, name: string, provinceId: number) {
    this.id = Number(id);
    this.name = name;
    this.provinceId = provinceId;
  }
}

File: app-api/Model/CityModel.cs
using app_api.Domain;

namespace app_api.Model
{
    public class CityModel
    {
        public CityModel(City city)
        {
            this.Id = city.Id;
            this.Name = city.Name;
            this.Province=new ProvinceModel(city.Province);
        }
        public long Id { get; set; }
        public ProvinceModel Province { get; set; }
        public string Name { get; set; }
    }
}
